api/views/slack_helpers.py

```diff
@@ -288,3 +290,29 @@ def is_valid_github_request(request: HttpRequest) -> bool:
 
     body_hex = f"sha1={body_hex}"
     return hmac.compare_digest(body_hex, github_signature)
+
+
+def is_valid_slack_request(request: HttpRequest) -> bool:
+    """Verify that a webhook from Slack is actually from them."""
+    # adapted from https://api.slack.com/authentication/verifying-requests-from-slack
+    if (slack_signature := request.headers.get("X-Slack-Signature")) is None:
+        return False
+
+    timestamp = request.headers["X-Slack-Request-Timestamp"]
+    if abs(time.time() - int(timestamp)) > 60 * 5:
+        # The request timestamp is more than five minutes from local time.
+        # It could be a replay attack, so let's ignore it.
+        return False
```

Comment:
Couldn't this header be modified by an attacker? Or is this integrity protected in some way?

Reply:
The timestamp is used as a part of the full computation, and the important bit here is that we have a very long key from Slack that's used when computing the sha256 hash. Folks can write the timestamp header, but they won't be able to compute the complete hash because that needs the Very Long Key:tm: that only us and Slack know.